test/ec2-meta-data-proxy/go.mod

```diff
@@ -0,0 +1,3 @@
+module github.com/aws/aws-node-termination-handler/ec2-meta-data-proxy
```

Comment:
Let's steer clear of Go submodules in the repo. multi-module repos are an extreme pain in the you-know-what. I've learned this from painful experience recently:

https://github.com/golang/go/wiki/Modules#should-i-have-multiple-modules-in-a-single-repository
https://github.com/golang/go/issues/35878

In short, if we used multiple Go modules in a single repo, we'd need to remember to apply a semver "prefixed" Git tag every time we issued a release, one for each Go submodule. It's a recipe for disaster and we can do just fine with a single semver Git-tagged repo with a single root Go.mod.